model/trigger/host.go

```diff
@@ -0,0 +1,169 @@
+package trigger
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/event"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/evergreen/model/notification"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	registry.registerEventHandler(event.ResourceTypeHost, event.EventHostExpirationWarningSent, makeHostTriggers)
+}
+
+const (
+	objectHost        = "host"
+	triggerExpiration = "expiration"
+
+	// notification templates
+	expiringHostTitle = `{{.Distro}} host termination reminder`
+	expiringHostBody  = `Your {{.Distro}} host with id {{.ID}} will be terminated at {{.ExpirationTime}}. Visit {{.URL}} to extend its lifetime.`
```

Comment:
for email, this needs to be a proper html document.

Reply:
Additionally, for slack can you move the Link to the Host page to a slack attachment?

Reply:
The previous version was a plain text email. I tested sending one to myself and it looked identical to the old notification. How do I make the link a slack attachment?

Reply:
Here is an example: https://github.com/evergreen-ci/evergreen/blob/master/model/trigger/patch.go#L155.
I recommend using the color "warning"